presto-main/src/main/java/com/facebook/presto/util/maps/IdentityLinkedHashMap.java

```diff
@@ -142,4 +133,116 @@ public int size()
             }
         };
     }
+
+    private class KeySet
+            extends SetView<K>
```

Comment:
Dear Reviewer, 

design decision here. I could extend `AbstractSet` and override only what needed (this is what `IdentityHashMap.EntrySet` does), but I find this wrong thing to do as long as I'm not part-time JDK collections dev. `AbstractSet` might be OK for well-behaved sets, which we're not, so it's not super obvious what needs to be overridden today and how that would work tomorrow.

Thus I traded code simplicity for code conciseness.